python/cudf/cudf/core/frame.py

```diff
@@ -1232,10 +1232,22 @@ def drop_duplicates(
         return result
 
     def replace(self, to_replace, replacement):
+        if isinstance(self, cudf.MultiIndex):
+            raise NotImplementedError(
+                "MultiIndex not yet supported for replace."
+            )
```

Comment:
Any reason MultiIndexes aren't supported yet? The codepath below is generalized for handling anything with `._data`

Reply:
It should work, but doesn't. I don't think it's a huge dive to solve but MI does not work out of the box when being run through the validation and find_and_replace. I've dug through a couple of the errors but don't have my head wrapped all the way around it yet. I'll ask in the chats today with further details as I get a better look and finish the other requests.

Reply:
OK, I fixed it. Pushing a new revision.

Reply:
Once this is merged lets just file an issue and tackle it in a follow up.